src/Features/Core/Portable/Completion/Providers/AbstractObjectInitializerCompletionProvider.cs

```diff
@@ -119,13 +119,47 @@ private static bool CanSupportObjectInitializer(ISymbol symbol)
 
         if (symbol is IFieldSymbol fieldSymbol)
         {
-            return !fieldSymbol.Type.IsStructType();
+            return MemberTypeCanSupportObjectInitializer(fieldSymbol.Type);
         }
         else if (symbol is IPropertySymbol propertySymbol)
         {
-            return !propertySymbol.Type.IsStructType();
+            return MemberTypeCanSupportObjectInitializer(propertySymbol.Type);
         }
 
         throw ExceptionUtilities.Unreachable();
     }
+
+    private static bool MemberTypeCanSupportObjectInitializer(ITypeSymbol type)
+    {
+        switch (type.SpecialType)
+        {
+            // While it is perfectly legal to assign instances of all these types to `{}`,
+            // it has no effects and is thus a needless recommendation
```

Comment:
i don't understand this comment.

Reply:
I mean assignments of the form `Property = {}` in the object initializer. Example of such a legal case:
```csharp
class C
{
    string S { get; }
}

new C()
{
    S = {},
};
```